src/blog/2024/08/opc-ua-to-mqtt-with-node-red.md

```diff
@@ -0,0 +1,135 @@
+
+### Transforming and Aggregating Data
+
+Once you have successfully retrieved data from your OPC UA server, the next step is to transform and aggregate this data to make it suitable for publishing to an MQTT broker. In this demonstration, we will aggregate the retrieved individual property values into a single object. Depending on your specific needs, you might choose to split the object properties and send them separately or perform various calculations and transformations on the data.
+
+1. Drag the Change node onto the canvas.
+2. Double-click on the node and set `msg.topic` to the name of the property you want to set for the retrieved data. In this context, set `msg.topic` to `'cycle-time'`, which will be the key in the object that we will create.
+
+![Setting the msg.topic with the Change node to retrieve data from the OPC UA server.](./images/change-node-setting-nodeid-datatype.png){data-zoomable}
+_Setting the msg.topic with the Change node to retrieve data from the OPC UA server._
+
+3. Drag the Join node onto the canvas. Set the mode to manual, with the option to create `msg.payload` using the values of `msg.topic` as keys. Set the count to 3 and ensure that the interval for all of the Inject nodes triggering data retrieval is the same. This ensures that the data is collected and aggregated correctly at the same time.
+4. Connect the output of the OPC UA client node (which retrieves the data) to the input of the Change node. For example, if I have set the Change node for the 'cycle-time' data property, connect it to the OPC UA client node that retrieves this data.
+5. Connect the output of the change node to the input of join node.
+6. Repeat this process for all of your data properties.
+
+### Sending Data to the MQTT Broker
+
+Now, in this section, we will show you how to send the collected data to an MQTT broker:
+
+1. Drag the mqtt out node onto the canvas.
+2. Double-click on it and configure it with your MQTT broker details. For this guide, I am utilizing the [free HiveMQ public broker](https://www.hivemq.com/mqtt/public-mqtt-broker/) which is specially desinged for testing and learning purpose.
+
+![Configuring the mqtt out node with broker information](./images/mqtt-out-node-config.png){data-zoomable}
+_Configuring the mqtt out node with broker information_
+
+3. Set the topic for your data in the MQTT Out node.
+4. Connect the output of the Join node to the input of the mqtt out node.
+5. Deploy the flow. After deploying, you will see the status "connected" with a green dot at the bottom of each node, indicating that you have successfully connected to your MQTT broker.
+
+### Retrieving Data from MQTT Broker
```

Comment:
This section feels incomplete and confusing.  We need to do the vice versa f the example from above.  We need to send data from MQTT to OPC.  Example: Sometimes, you will need to send data to an OPC server.  This can be used to define recipes from MES or some other external system.

Reply:
So, this section should cover MQTT to OPC-UA rather than retrieval operations, correct?

Reply:
Yes.  That or drop the section altogether.